File: Algorithms/remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        c = 0
        r = ""
        for i in S:
            if i == "(":
                c += 1
                if c > 1:
                    r += i
                    continue
            if i == ")":
                c -= 1
                if c > 0:
                    r += i
                    continue
        return r

File: Algorithms/remove-outermost-parentheses.go
package solution

func removeOuterParentheses(S string) string {
    c := 0
    r := ""
    for _, rune := range S {
        if rune == '(' {
            c++
            if c > 1 {
                r += string(rune)
            }
            continue
        } else {
            c--
            if c > 0 {
                r += string(rune)
            }
            continue
        }
    }
    return r
}